platform/kobo/koreader.sh

```diff
@@ -2,7 +2,7 @@
 export LC_ALL="en_US.UTF-8"
 
 # working directory of koreader
-KOREADER_DIR="${0%/*}"
+KOREADER_DIR="$(CDPATH= cd -- "$(dirname -- "$0")" && pwd -P)"
```

Comment:
I don't quite follow the logic here? Maybe it's too late. If I've got it right:

* Our koreader.sh here is probably `$0` (for example /mnt/onboard/.adds/koreader/koreader.sh)
* `dirname` makes that `/mnt/onboard/.adds/koreader`
* So only if those commands were successful, do `pwd -P`? Is there some kind of check missing here? Wouldn't that first part always be successful?

Reply:
The idea is to only print the (absolute) path (via `pwd`) if we can successfully reach it. Given the fact that we know the shell is going to be busybox ash and that no-one will probably ever be attempting to source us, there's not much reason for the cd to fail, but the idea was to be extremely defensive and catch degenerate command lines early ;).

We also doublez-check do that by aborting if we can't cd to `KOREADER_DIR` below.

Reply:
In the same vein, the `-P` is overkill, because vfat makes it highly unlikely for us to encounter a symlink ^^.

Reply:
c.f., https://stackoverflow.com/a/29835459